File: 프로그래머스/0/120905. n의 배수 고르기/n의 배수 고르기.java
class Solution {
    public int[] solution(int n, int[] numlist) {
        int num = 0;
        
        for(int i = 0; i < numlist.length; i++) {
            if(numlist[i] % n == 0) {
                num++;
            }
        }
        
        int answer[] = new int[num];
        int index = 0;
        
        for(int i = 0; i < numlist.length; i++) {
            if(numlist[i] % n == 0) {
                answer[index] = numlist[i];
                index++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120905. n의 배수 고르기/n의 배수 고르기.py
def solution(n, numlist):
    answer = []
    
    # 1.
    # for i in numlist:
    #     if i % n == 0:
    #         answer.append(i)
    
    # 2. 한번에 쓰기
    answer = [i for i in numlist if i % n == 0]
    
    return answer